cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/bulkwriter/Tokenizer.java

```diff
@@ -39,13 +38,12 @@ interface SerializableFunction<T, R> extends Function<T, R>, Serializable
     private final TokenUtils tokenUtils;
     private final SerializableFunction<Object[], Object[]> keyColumnProvider;
 
-    public Tokenizer(BulkWriterContext writerContext)
+    public Tokenizer(BroadcastableTableSchema broadcastableTableSchema, boolean isMurmur3Partitioner)
     {
-        TableSchema tableSchema = writerContext.schema().getTableSchema();
-        this.tokenUtils = new TokenUtils(tableSchema.partitionKeyColumns,
-                                         tableSchema.partitionKeyColumnTypes,
-                                         writerContext.cluster().getPartitioner() == Partitioner.Murmur3Partitioner);
-        this.keyColumnProvider = tableSchema::getKeyColumns;
+        this.tokenUtils = new TokenUtils(broadcastableTableSchema.getPartitionKeyColumns(),
+                                         broadcastableTableSchema.getPartitionKeyColumnTypes(),
+                                         isMurmur3Partitioner);
```

Comment:
it seems a little silly that we pass a boolean flag here for determining whether the partitioner is murmur3 or not. Can't we just directly pass the partitioner to TokenUtils instead?

Reply:
Passing the boolean because that is wanted by `TokenUtils` constructor. I would keep it as-is. The change would be unrelated to this refactor. 

```java
    public TokenUtils(List<String> partitionKeyColumns,
                      List<ColumnType<?>> partitionKeyColumnTypes,
                      boolean isMurmur3Partitioner)
```